servicelayer/jobs.py

```diff
@@ -44,6 +86,48 @@ def remove_dataset(cls, conn, dataset):
             job.remove()
 
 
+class Task(object):
+    def __init__(self, payload, context, job_stage):
+        self.payload = payload
+        self.context = context
+        self.stage = job_stage
+        self.task_id = self._make_task_id()
+
+    def serialize(self):
+        return dump_json({
+            'context': self.context or {},
+            'payload': self.payload,
+            'dataset': self.stage.dataset,
+            'job_id': self.stage.job.id,
+            'stage': self.stage.stage,
+            'priority': self.stage.priority
+        })
+
+    def queue(self):
+        self.stage.queue_task(self)
+
+    def _make_task_id(self):
```

Comment:
This seems super risky and computationally intense to me. Should we not just generate a guid when the task is first scheduled that is guaranteed to be unique, rather than relying on hashing a potentially complex data structure?

Reply:
We are hashing a string which should not be that bad. But yes, uuid is definitely better.